llvm-spirv/lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -2244,7 +2244,10 @@ SPIRVValue *LLVMToSPIRV::transIntrinsicInst(IntrinsicInst *II,
     }
     Value *Len = MSI->getLength();
     if (!isa<ConstantInt>(Len)) {
-      assert(!"Can't translate llvm.memset with non-const `length` argument");
+      BM->getErrorLog().checkError(
+          isa<ConstantInt>(Len), SPIRVEC_InvalidFunctionCall,
+          toString(II) +
+              "\nCan't translate llvm.memset with non-const `length` argument");
```

Comment:
I would change the diagnostic text to something like:
"\nTranslation of llvm.memset requires a non-const `length` argument"

to match other diagnostic formats used in this file.